src/samplers/aliastable.jl

```diff
@@ -1,21 +1,21 @@
-struct AliasTable{S} <: Sampleable{Univariate,Discrete}
-    accept::Vector{Float64}
-    alias::Vector{Int}
-    isampler::S
+struct AliasTable{N} <: Sampleable{Univariate,Discrete}
+    accept::NTuple{N,Float64}
+    alias::NTuple{N,Int}
+    sample_space::UnitRange{Int64}
 end
-ncategories(s::AliasTable) = length(s.accept)
+ncategories(s::AliasTable{N}) where N = N
 
 function AliasTable(probs::AbstractVector{T}) where T<:Real
-    n = length(probs)
-    n > 0 || throw(ArgumentError("The input probability vector is empty."))
-    accp = Vector{Float64}(undef, n)
-    alias = Vector{Int}(undef, n)
+    N = length(probs)
+    N > 0 || throw(ArgumentError("The input probability vector is empty."))
+    accp = Vector{Float64}(undef, N)
+    alias = Vector{Int}(undef, N)
     StatsBase.make_alias_table!(probs, 1.0, accp, alias)
-    AliasTable(accp, alias, Random.RangeGenerator(1:n))
+    AliasTable{N}(tuple(accp...), tuple(alias...), 1:N)
 end
 
 function rand(rng::AbstractRNG, s::AliasTable)
-    i = rand(rng, s.isampler) % Int
+    i = rand(rng, Random.Sampler(rng, s.sample_space)) % Int
```

Comment:
So a `Sampler` is best explicitly used typically when called repeatedly. Here, this is strictly equivalent to `i = rand(rng, s.sampler_space)`.

Reply:
dropped use of `Sampler` in favour of direct sampling